src/diffusers/utils/hub_utils.py

```diff
@@ -359,3 +367,95 @@ def _get_model_file(
                 f"Otherwise, make sure '{pretrained_model_name_or_path}' is the correct path to a directory "
                 f"containing a file named {weights_name}"
             )
+
+
+# Adapted from
+# https://github.com/huggingface/transformers/blob/5bb4430edc7df9f9950d412d98bbe505cc4d328b/src/transformers/utils/hub.py#L628
+class PushToHubMixin:
+    """
+    A Mixin containing the functionality to push a model/scheduler to the Hugging Face Hub.
+    """
+
+    def _upload_modified_files(
+        self,
+        working_dir: Union[str, os.PathLike],
+        repo_id: str,
+        commit_message: Optional[str] = None,
+        create_pr: bool = False,
+    ):
+        """
+        Uploads all modified files in `working_dir` to `repo_id`.
+        """
+        if commit_message is None:
+            if "Model" in self.__class__.__name__:
+                commit_message = "Upload model"
+            elif "Scheduler" in self.__class__.__name__:
+                commit_message = "Upload scheduler"
+            else:
+                commit_message = f"Upload {self.__class__.__name__}"
+
+        logger.info(f"Uploading the files of {working_dir} to {repo_id}.")
+        return upload_folder(
+            repo_id=repo_id, folder_path=working_dir, commit_message=commit_message, create_pr=create_pr
+        )
```

Comment:
I don't think defining this method is useful. The only thing it does in addition to `upload_folder` is to define a default commit message. If this is really necessary, I would define a `_get_commit_message(self, commit_message: str | None) -> str` that would do the trick but that's all. Otherwise I found it a bit redundant and misleading (the method is called `_upload_modified_files` but is actually uploading everything).